### PR TITLE
MXNet benchmark scripts

### PR DESCRIPTION
**Pull request by nswamy**
Scripts to run MXNet benchmark on popular image networks.

### CODE REVIEW

File: example/image-classification/train_model.py [hunk 1]
```diff
@@ -83,22 +83,31 @@ def fit(args, network, data_loader, batch_end_callback=None):
...
-        eval_metric        = eval_metrics,
-        kvstore            = kv,
-        batch_end_callback = batch_end_callback,
-        epoch_end_callback = checkpoint)
+    if args.benchmark:
+        batch_end_callback.append(mx.callback.Speedometer(args.batch_size, 10))
+        # don't run evaluation for benchmark
+        model.fit(
+            X                  = train,
+            eval_data          = val,
```
Comment (piiswrong) [inline example/image-classification/train_model.py]: Isn't this still running eval?

Comment (nswamy) [inline example/image-classification/train_model.py]: eval_metrics are not passed and also the batch_end_callback is not, so I am assuming it won't.

Comment (piiswrong) [inline example/image-classification/train_model.py]: Please confirm.  I think the current script will break.
you need to not pass eval_data to skip validation.

Comment (piiswrong) [inline example/image-classification/train_model.py]: That means evaluation is still running but the acc was not logged. Is that desired behavior?

**Comment by nswamy** [comment]
Yes that should be fine, I'll consult Mu and make separate pull request if that needs to change.

**Comment by ChasonLee** [comment]
@nswamy Now there is an inconsistency between **train_mnist.py** and **train_model.py**.

**Comment by nswamy** [comment]
@ChasonLee, what is the inconsistency ? Is it  **train_mnist.py** does not have an option to benchmark ? 
Please elaborate.

**Comment by ChasonLee** [comment]
@nswamy Yes, could you PR again to fix it?

**Comment by nswamy** [comment]
Do people really care about benchmarking **mnist** ? 
sure why not, I will make a separate PR to bring them in sync.

**Comment by ChasonLee** [comment]
I am sorry to make you misunderstand, I am not saying benchmarking mnist. I just want to know, is there a way to add your benchmark without affecting the normal use of train_mnist.py.

**Pull request merged**